src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,51 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl?.split('/'))?.split('.')?.join('.')
+      // check if file can just be fetched without sas token
+      try {
+        const res = await fetchOk(azureStorageUrl)
+        const text = await res.headers.entries()
+        const dict = {}
+        for (const pair of text) {
+          dict[pair[0]] = pair[1]
+        }
+
+        return { lastModified: dict['last-modified'], size: dict['content-length'], azureStorageUrl, workspaceId, fileName }
```

Comment:
This code is repeated twice in this component (again down below in this function) -- would it make sense to move it into a helper function?

Reply:
moved everything but the fetch into a function to handle headers specifically - I was thinking that if in the future we wanted to do something with the response body itself, it may be nice to have that inside the main function but I could be convinced otherwise. let me know what you think!